### PR TITLE
Added history calculation for all the views

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
I've added the history calculation for all the views. Let me know what you think about it thank you. If this feature can updated in some way you can give me instructions and I'll try to do it.

### CODE REVIEW

**Review by AnthonyH93 — CHANGES_REQUESTED** [review]
Looks very good thanks! Please remove the `.DS_Store` file though since it is causing a merge conflict.

**Comment by AnthonyH93** [comment]
In terms of future changes, I would have the following suggestions:
* Technical
  * Add the result of the computation to the table
    * Maybe as part of the same string with an equals sign separator (ex. 6 * 5 = 30)
    * Or as a new column (ex. 6 * 5 | 30)
  * Allow a long press or single click to copy answers from the table 
* Visual
  * Make the background of the calculation history overlay black
  * Make the text of the overlay either white or the colour theme preference
  * Make the navigation text in the corner to open the calculation history overlay the colour theme preference

Feel free to suggest otherwise if you don't think these changes make sense. Again, thanks a lot for the help on this!

Also, do you think it would be good for the history to persist after closing the app or should it only store the current session calculations?

**Comment by FilippoZazzeroni** [comment]
I will try to implement the things you suggested, that for me all of them make sense. The only thing I don't understand is the copy of the history content if it can be pasted in the calculator later or if that is the purpose. The history so far, I would be keep it run time there s no need to persist it.

**Comment by AnthonyH93** [comment]
Thanks for implementing those features. I'll run the app soon and review. Yes, the purpose of copying would be to paste into the calculator later, I suppose we could skip the step of copying and just have it automatically close the overlay and put the number into the calculator when you click on a row, but copying is probably good for now.

**Comment by FilippoZazzeroni** [comment]
I've already added the copy paste functionality when you click a row. Let me know how is when you try it.

File: HexaCalc/View Controllers/CalculationHistoryViewController.swift [hunk 1]
```diff
@@ -34,15 +38,31 @@ extension CalculationHistoryViewController: UITableViewDelegate, UITableViewData
...
         }
         
         return cell
     }
     
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let uiPasteboard = UIPasteboard.general
+        uiPasteboard.string = calculationHistory[indexPath.row].result
+        
+        let alert = UIAlertController(title: "Result copied to clipboard", message: nil, preferredStyle: .alert)
```
Comment (AnthonyH93) [inline HexaCalc/View Controllers/CalculationHistoryViewController.swift:58]:
```suggestion
        let alert = UIAlertController(title: "Copied to Clipboard", message: "\(calculationHistory[indexPath.row].result) has been added to your clipboard.", preferredStyle: .alert)
```

File: HexaCalc/View Controllers/CalculationHistoryViewController.swift [hunk 2]
```diff
@@ -34,15 +38,31 @@ extension CalculationHistoryViewController: UITableViewDelegate, UITableViewData
...
+        alert.view.tintColor = loadPrefrredColorFromPreferences()
+        alert.addAction(UIAlertAction(title: "OK", style: .cancel))
```
Comment (AnthonyH93) [inline HexaCalc/View Controllers/CalculationHistoryViewController.swift:60]:
```suggestion
```

File: HexaCalc/View Controllers/CalculationHistoryViewController.swift [hunk 3]
```diff
@@ -34,15 +38,31 @@ extension CalculationHistoryViewController: UITableViewDelegate, UITableViewData
...
+        alert.overrideUserInterfaceStyle = .dark
+
```
Comment (AnthonyH93) [inline HexaCalc/View Controllers/CalculationHistoryViewController.swift:62]:
```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
            alert.dismiss(animated: true, completion: nil)
        }
```

**Review by AnthonyH93 — CHANGES_REQUESTED** [review]
Please change the copy alert to match the other copy alerts (sorry this one is my fault, I should have a well defined module for copy alerts). Also please delete `.DS_Store` since it is causing a merge conflict.

Other than that it looks awesome, thanks a lot!

**Review by AnthonyH93 — APPROVED** [review]
Looks great, thanks for making the requested changes!

**Pull request merged**